python/fbprophet/forecaster.py

```diff
@@ -450,13 +447,22 @@ def make_holiday_features(self, dates):
         all_holidays = self.holidays
         if self.append_holidays is not None:
             year_list = list({x.year for x in dates})
-            append_holidays_df = self.make_holidays(
+            append_holidays_df = make_holidays(
                                     year_list=year_list,
                                     country_list=self.append_holidays)
+            #Validates names of generated holidays.
+            if self.train_holiday_names is None:
+                for name in append_holidays_df.holiday:
+                    if name in self.seasonalities:
+                        raise ValueError(
+                            'Seasonality name "{}" already used for a holiday.'.format(name))
+                    if name in self.extra_regressors:
+                        raise ValueError(
+                            'Regressor name "{}" already used for a holiday.'.format(name))
```

Comment:
This block from L453 to L461 needs to be removed. We want to validate names when seasonalities/regressors are added, not during fitting. (At this point it is too late to change the name without creating a whole new model).

What we ideally want is:
* In the make_holidays.py, make a function `get_holiday_names(append_holidays)` that returns a set of all holiday names that could possibly be returned for the holidays in append_holidays. Import that function here.
* In L227 here, you can see where there is an if block for validating that the name doesn't overlap with a holiday name. Add a block right below that one to validate that it doesn't overlap with an append_holidays name, like this:
```
if (check_holidays and self.append_holidays is not None and
        name in get_holiday_names(self.append_holidays
):
    raise ValueError(
        'Name "{}" already used for a built-in holiday.'.format(name)
    )
```